features/roomlist/src/test/kotlin/io/element/android/features/roomlist/RoomListPresenterTests.kt

```diff
@@ -14,34 +14,180 @@
+            assertThat(withRoomState.roomList.size).isEqualTo(1)
+            assertThat(withRoomState.roomList.first()).isEqualTo(aRoomListRoomSummary)
+        }
+    }
+
+    @Test
+    fun `present - load 1 room with success and filter rooms`() = runTest {
+        val roomSummaryDataSource = InMemoryRoomSummaryDataSource()
+        val presenter = RoomListPresenter(
+            FakeMatrixClient(
+                sessionId = SessionId("sessionId"),
+                roomSummaryDataSource = roomSummaryDataSource
+            ),
+            createDateFormatter()
+        )
+        moleculeFlow(RecompositionClock.Immediate) {
+            presenter.present()
+        }.test {
+            roomSummaryDataSource.postRoomSummary(listOf(aRoomSummaryFilled()))
+            skipItems(3)
+            val loadedState = awaitItem()
+            // Test filtering with result
+            loadedState.eventSink.invoke(RoomListEvents.UpdateFilter(A_ROOM_NAME.substring(0, 3)))
+            val withNotFilteredRoomState = awaitItem()
+            assertThat(withNotFilteredRoomState.filter).isEqualTo(A_ROOM_NAME.substring(0, 3))
+            assertThat(withNotFilteredRoomState.roomList.size).isEqualTo(1)
+            assertThat(withNotFilteredRoomState.roomList.first()).isEqualTo(aRoomListRoomSummary)
+            // Test filtering without result
+            withNotFilteredRoomState.eventSink.invoke(RoomListEvents.UpdateFilter("tada"))
+            skipItems(1) // Filter update
+            val withFilteredRoomState = awaitItem()
+            assertThat(withFilteredRoomState.filter).isEqualTo("tada")
+            assertThat(withFilteredRoomState.roomList.size).isEqualTo(0)
+        }
+    }
+
+    @Test
+    fun `present - update visible range`() = runTest {
+        val roomSummaryDataSource = InMemoryRoomSummaryDataSource()
+        val presenter = RoomListPresenter(
+            FakeMatrixClient(
+                sessionId = SessionId("sessionId"),
+                roomSummaryDataSource = roomSummaryDataSource
+            ),
+            createDateFormatter()
+        )
+        moleculeFlow(RecompositionClock.Immediate) {
+            presenter.present()
+        }.test {
+            roomSummaryDataSource.postRoomSummary(listOf(aRoomSummaryFilled()))
+            skipItems(3)
+            val loadedState = awaitItem()
+            // check initial value
+            assertThat(roomSummaryDataSource.latestSlidingSyncRange).isNull()
+            // Test empty range
+            loadedState.eventSink.invoke(RoomListEvents.UpdateVisibleRange(IntRange(1, 0)))
+            assertThat(roomSummaryDataSource.latestSlidingSyncRange).isNull()
+            // Update visible range and check that range is transmitted to the SDK after computation
+            loadedState.eventSink.invoke(RoomListEvents.UpdateVisibleRange(IntRange(0, 0)))
+            assertThat(roomSummaryDataSource.latestSlidingSyncRange).isEqualTo(IntRange(0, 20))
```

Comment:
Not sure it's really useful to test that, because if the algorithm change it will break the test.

Reply:
> if the algorithm change it will break the test

this is what tests are for, no? If the algorithm is updated without the desire to change its behavior, the tests will check that.